src/pages/VaccinePage.js

```diff
@@ -1,125 +1,196 @@
-  return {
-    isAdmin,
-    minLine,
-    maxLine,
-    minDomain,
-    maxDomain,
-    fridges,
-    selectedFridge,
-    breaches,
-  };
+FridgeDisplay.defaultProps = {
+  fridge: null,
+};
+FridgeDisplay.propTypes = {
+  fridge: PropTypes.shape({
+    id: PropTypes.string,
+    description: PropTypes.string,
+    sensors: PropTypes.object,
+  }),
+  blinkSensor: PropTypes.func.isRequired,
+  toFridgeDetail: PropTypes.func.isRequired,
+  toEditSensorPage: PropTypes.func.isRequired,
 };
 
-const mapDispatchToProps = dispatch => ({
-  onViewAdminPage: () => dispatch(gotoVaccineAdminPage()),
-  onSelectFridge: fridge => dispatch(FridgeActions.select(fridge)),
-  onOpenBreachModal: breachId => {
-    dispatch(BreachActions.viewFridgeBreach(breachId));
-  },
-});
-
+export const VaccinePageComponent = ({ fridges, toNewSensorPage, ...dispatchers }) => (
+  <DataTablePageView style={{ paddingHorizontal: 20, paddingVertical: 30 }}>
+    <FlexRow style={{ justifyContent: 'flex-end', marginBottom: 10 }}>
+      <PageButton text={buttonStrings.add_sensor} onPress={toNewSensorPage} />
+    </FlexRow>
+    <AfterInteractions placeholder={null}>
+      <Animatable.View animation="fadeIn" duration={500} useNativeDriver>
+        <FlatList
+          renderItem={({ item }) => <FridgeDisplay fridge={item} {...dispatchers} />}
+          data={fridges}
+        />
+      </Animatable.View>
+    </AfterInteractions>
+  </DataTablePageView>
+);
 VaccinePageComponent.defaultProps = {
-  breaches: [],
-  selectedFridge: null,
+  fridges: null,
 };
-
 VaccinePageComponent.propTypes = {
-  selectedFridge: PropTypes.object,
-  minLine: PropTypes.array.isRequired,
-  maxLine: PropTypes.array.isRequired,
-  breaches: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
-  maxDomain: PropTypes.number.isRequired,
-  minDomain: PropTypes.number.isRequired,
-  fridges: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
-  onSelectFridge: PropTypes.func.isRequired,
-  onOpenBreachModal: PropTypes.func.isRequired,
-  onViewAdminPage: PropTypes.func.isRequired,
-  isAdmin: PropTypes.bool.isRequired,
+  fridges: PropTypes.object,
+  blinkSensor: PropTypes.func.isRequired,
+  toFridgeDetail: PropTypes.func.isRequired,
+  toEditSensorPage: PropTypes.func.isRequired,
+  toNewSensorPage: PropTypes.func.isRequired,
 };
 
 const localStyles = StyleSheet.create({
-  blankText: { textAlign: 'center', fontFamily: APP_FONT_FAMILY, fontSize: 20 },
+  iconButton: {
+    flexBasis: 50,
+    justifyContent: 'center',
+  },
+  fridgeDetail: {
+    flex: 0,
+    height: 60,
+    paddingHorizontal: 10,
+  },
+  fridgePaperContentContainer: {
+    flex: 2,
+    flexDirection: 'column',
+  },
+});
+
+const stateToProps = state => {
+  const { fridge } = state;
+  const { fridges } = fridge;
+  return { fridges };
+};
+
+const dispatchToProps = dispatch => ({
+  blinkSensor: macAddress => dispatch(VaccineActions.startSensorBlink(macAddress)),
```

Comment:
Oh. Think I missed updating this with https://github.com/openmsupply/mobile/pull/3411 to `BlinkActions.startSensorBlink`. I suppose can be slotted in with #3428 though.

Reply:
b466ddc